syweb/webclient/app/settings/settings-controller.js

```diff
@@ -264,25 +317,58 @@ function($scope, matrixService, modelService, eventHandlerService, notificationS
         setMuteNotifications(!$scope.config.muteNotifications);
     };
 
-    $scope.addContentRule = function(pattern, actions) {
-        notificationService.addGlobalContentRule(pattern, actions).then(function() {
+    $scope.addContentRule = function() {
+        var actions = [ $scope.rule_add_action.content ];
+        if ($scope.rule_add_sound.content) {
+            actions.push({'set_tweak': 'sound', 'value': 'default'});
+        }
+        actions.push({'set_tweak': 'highlight', 'value': $scope.rule_highlight.content});
+        notificationService.addGlobalContentRule($scope.rule_add_input.content, actions).then(function() {
+            $scope.rule_add_inprogress.content = false;
             notificationService.clearRulesCache();
             fetchRules();
+            $scope.rule_add_input.content = '';
+        }, function() {
+            $scope.rule_add_inprogress.content = false;
+            $scope.feedback = "Failed to add rule";
         });
+        $scope.rule_add_inprogress.content = true;
     };
 
-    $scope.addRoomRule = function(room_id, actions) {
-        notificationService.addGlobalRoomRule(room_id, actions).then(function() {
+    $scope.addRoomRule = function() {
+        var actions = [ $scope.rule_add_action.room ];
+        if ($scope.rule_add_sound.room) {
+            actions.push({'set_tweak': 'sound', 'value': 'default'});
+        }
+        actions.push({'set_tweak': 'highlight', 'value': $scope.rule_highlight.room});
+        notificationService.addGlobalRoomRule($scope.rule_add_input.room, actions).then(function() {
+            $scope.rule_add_inprogress.room = false;
             notificationService.clearRulesCache();
             fetchRules();
+            $scope.rule_add_input.room = '';
+        }, function() {
+            $scope.rule_add_inprogress.room = false;
+            $scope.feedback = "Failed to add rule";
```

Comment:
There's a fair amount of generic error handling which is duplicated for each rule type (see ``:368`` and ``:333`` for examples). Is it possible to factor this out a bit?